File: app/src/main/java/org/neidhardt/dynamicsoundboard/views/BaseDialog.java
package org.neidhardt.dynamicsoundboard.views;

import android.app.DialogFragment;
import android.os.Bundle;
import org.neidhardt.dynamicsoundboard.DynamicSoundboardApplication;
import org.neidhardt.dynamicsoundboard.soundactivity.SoundActivity;
import org.neidhardt.dynamicsoundboard.soundmanagement.model.SoundsDataAccess;
import org.neidhardt.dynamicsoundboard.soundmanagement.model.SoundsDataStorage;
import org.neidhardt.dynamicsoundboard.soundsheetmanagement.model.SoundSheetsDataAccess;
import org.neidhardt.dynamicsoundboard.soundsheetmanagement.model.SoundSheetsDataStorage;
import org.neidhardt.dynamicsoundboard.soundsheetmanagement.model.SoundSheetsDataUtil;

import javax.inject.Inject;

/**
 * File created by eric.neidhardt on 14.11.2014.
 */
public abstract class BaseDialog extends DialogFragment
{
	@Inject protected SoundSheetsDataAccess soundSheetsDataAccess;
	@Inject protected SoundSheetsDataStorage soundSheetsDataStorage;
	@Inject protected SoundSheetsDataUtil soundSheetsDataUtil;

	@Inject protected SoundsDataStorage soundsDataStorage;
	@Inject protected SoundsDataAccess soundsDataAccess;

	private DialogBaseLayout mainView;

	public DialogBaseLayout getMainView()
	{
		return mainView;
	}

	public void setMainView(DialogBaseLayout mainView)
	{
		this.mainView = mainView;
	}

	@Override
	public void onCreate(Bundle savedInstanceState)
	{
		super.onCreate(savedInstanceState);
		DynamicSoundboardApplication.getApplicationComponent().inject(this);
	}

	public SoundActivity getSoundActivity()
	{
		return (SoundActivity) this.getActivity();
	}
}

File: app/src/main/kotlin/org/neidhardt/dynamicsoundboard/views/BaseDialog.kt
package org.neidhardt.dynamicsoundboard.views

import android.support.v4.app.DialogFragment
import org.neidhardt.dynamicsoundboard.SoundboardApplication
import org.neidhardt.dynamicsoundboard.soundactivity.SoundActivity
import org.neidhardt.dynamicsoundboard.soundlayoutmanagement.model.SoundLayoutsAccess
import org.neidhardt.dynamicsoundboard.soundmanagement.model.SoundsDataAccess
import org.neidhardt.dynamicsoundboard.soundmanagement.model.SoundsDataStorage
import org.neidhardt.dynamicsoundboard.soundsheetmanagement.model.SoundSheetsDataAccess
import org.neidhardt.dynamicsoundboard.soundsheetmanagement.model.SoundSheetsDataStorage
import org.neidhardt.dynamicsoundboard.soundsheetmanagement.model.SoundSheetsDataUtil

/**
 * File created by eric.neidhardt on 14.11.2014.
 */
abstract class BaseDialog : DialogFragment()
{
	protected val KEY_CALLING_FRAGMENT_TAG: String = "KEY_CALLING_FRAGMENT_TAG"

	protected var soundSheetsDataAccess: SoundSheetsDataAccess = SoundboardApplication.soundSheetsDataAccess
	protected var soundSheetsDataStorage: SoundSheetsDataStorage = SoundboardApplication.soundSheetsDataStorage
	protected var soundSheetsDataUtil: SoundSheetsDataUtil = SoundboardApplication.soundSheetsDataUtil

	protected var soundsDataStorage: SoundsDataStorage = SoundboardApplication.soundsDataStorage
	protected var soundsDataAccess: SoundsDataAccess = SoundboardApplication.soundsDataAccess

	var soundLayoutsAccess: SoundLayoutsAccess = SoundboardApplication.soundLayoutsAccess

	val soundActivity: SoundActivity
		get() = this.activity as SoundActivity
}

